core/types/transaction.go

```diff
@@ -784,6 +787,23 @@ type MevBundle struct {
 	Hash              common.Hash
 }
 
+func (b *MevBundle) UniquePayload() []byte {
+	var buf []byte
+	buf = binary.AppendVarint(buf, b.BlockNumber.Int64())
+	buf = append(buf, b.Hash[:]...)
+	sort.Slice(b.RevertingTxHashes, func(i, j int) bool {
+		return bytes.Compare(b.RevertingTxHashes[i][:], b.RevertingTxHashes[j][:]) <= 0
+	})
+	for _, txHash := range b.RevertingTxHashes {
+		buf = append(buf, txHash[:]...)
+	}
+	return buf
+}
+
+func (b *MevBundle) ComputeUUID() uuid.UUID {
+	return uuid.NewHash(sha3.NewLegacyKeccak256(), uuid.Nil, b.UniquePayload(), 5)
```

Comment:
question: Why are you using legacy keccak here?

Reply:
I was thinking among the lines of unity with existing geth codebase/possible js tooling. If not for this I would just go for sha256.

Reply:
yes, please use sha256 or blake2b, I don't think we need legacy keccak here - that's intended for very specific case